db/mysql/ops.go

```diff
@@ -0,0 +1,264 @@
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	domain "github.com/Hammad887/chat-app/models"
+	"github.com/dgrijalva/jwt-go"
+	"github.com/google/uuid"
+	"golang.org/x/crypto/bcrypt"
+)
+
+func (c *client) RegisterUser(ctx context.Context, user *domain.User) (bool, error) {
+
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(user.Password), bcrypt.DefaultCost)
+	if err != nil {
+		return false, err
+	}
+
+	user.ID = uuid.New().String()
+
+	if _, err := c.dbc.Exec("INSERT INTO users (id, name, email, password) VALUES (?, ?, ?, ?)", user.ID, user.Name, user.Email, string(hashedPassword)); err != nil {
+		return false, err
+
+	}
+
+	// createChatRoom()
+	return true, nil
+}
+
+func (c *client) SendMessage(ctx context.Context, id string, message *domain.Message) (bool, error) {
```

Comment:
why using bool here, you can achieve this without error return type only

Reply:
Changed